cats/__init__.py

```diff
@@ -51,8 +113,55 @@ def parse_arguments():
              "Default: if absent, the total carbon footprint is not estimated."
     )
 
+    parser.add_argument("--format", type=str, help="Format to output optimal start time and carbon emmission"
+                        "estimate savings in. Currently only JSON is supported.", choices=["json"])
+
     return parser
 
+@dataclasses.dataclass
+class CATSOutput:
+    """Carbon Aware Task Scheduler output"""
+
+    carbonIntensityAPI: str
+    carbonIntensityNow: CarbonIntensityAverageEstimate
+    carbonIntensityOptimal: CarbonIntensityAverageEstimate
+    location: str
+    countryISO3: str
+    emmissionEstimate: Optional[Estimates] = None
+
+    def __str__(self) -> str:
+        out = f"Best job start time: {self.carbonIntensityOptimal.start}"
+
+        if self.emmissionEstimate:
+            out += (f"Estimated emmissions for running job now: {self.emmissionEstimate.now}\n"
+                f"Estimated emmissions for running delayed job: {self.emmissionEstimate.best})\n"
+                f" (- {self.emmissionEstimate.savings})"
+            )
+        return out
+
+    def to_json(self, dateformat=None, **kwargs) -> str:
+        if "%" not in dateformat:
```

Comment:
This raises a `TypeError` if `dateformat` is `None` (can't iterate on `None`).

What about processing the `dateformat` argument before calling `to_json` down [here](https://github.com/GreenScheduler/cats/pull/63/files#diff-80d0db763b62024ceb8b4c93581731d93de38ae20a5bc7d0d35c4ada7ece9b5cR264)

Reply:
Thanks! done